IOPool/Output/src/RootOutputFile.cc

```diff
@@ -696,6 +694,27 @@ namespace edm {
 
     std::set<StoredProductProvenance> provenanceToKeep;
 
+    ProductProvenanceRetriever const* provRetriever = nullptr;
+    {
+      EventPrincipal const* eventPrincipal = dynamic_cast<EventPrincipal const*>(&principal);
+      if (eventPrincipal) {
+        provRetriever =eventPrincipal->productProvenanceRetrieverPtr();
+        assert(provRetriever);
+      }
+    }
+    
+    //If we are dropping some of the meta data we need to know
```

Comment:
It is there to replace the call on line 675 that I removed 
`principal.getProvenance(info->branchID(), mcc).product().produced())`

That `getProvenance` call required access to the principal and could trigger reading from the file but that same information can be gotten from the `ProductRegistry`.